modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.svg4everybody.web.internal.servlet.taglib;
+
+import com.liferay.portal.kernel.servlet.BrowserSniffer;
+import com.liferay.portal.kernel.servlet.taglib.BaseDynamicInclude;
+import com.liferay.portal.kernel.servlet.taglib.DynamicInclude;
+import com.liferay.portal.url.builder.AbsolutePortalURLBuilder;
+import com.liferay.portal.url.builder.AbsolutePortalURLBuilderFactory;
+import com.liferay.portal.util.PropsValues;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Bryce Osterhaus
+ */
+@Component(
+	immediate = true, property = "service.ranking:Integer=" + Integer.MAX_VALUE,
+	service = DynamicInclude.class
+)
+public class SVG4EverybodyTopHeadDynamicInclude extends BaseDynamicInclude {
+
+	@Override
+	public void include(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, String key)
+		throws IOException {
+
+		if (!PropsValues.CDN_DYNAMIC_RESOURCES_ENABLED ||
```

Comment:
Shouldn't this be `if (PropsValues.CDN_DYNAMIC_RESOURCES_ENABLED)` instead of the negated version? 🤔

Reply:
Reading through the docs, it says:

```
    #
    # Set this to true to enable serving dynamically generated CSS, JavaScript,
    # and images via a CDN. Setting this to false allows the usage of CDNs that
    # do not support lazy loading of resources (e.g. Amazon CloudFront).
    #
    # Env: LIFERAY_CDN_PERIOD_DYNAMIC_PERIOD_RESOURCES_PERIOD_ENABLED
    #
    cdn.dynamic.resources.enabled=true
```

And we also have `LIFERAY_CDN_PERIOD_HOST_PERIOD_HTTP` and `LIFERAY_CDN_PERIOD_HOST_PERIOD_HTTPS`, so maybe we need a combination of these?

If I'm reading this correctly, one would need to set this propery to `false` now (which is different from the default) to benefit from this.

Maybe we want something like:

```java
PropsValues.CDN_DYNAMIC_RESOURCES_ENABLED && (PropsValues.LIFERAY_CDN_PERIOD_HOST_PERIOD_HTTP || PropsValues.LIFERAY_CDN_PERIOD_HOST_PERIOD_HTTPS)
```

Reply:
I see that based on https://issues.liferay.com/browse/LPS-71763, the steps to enable CDN are:

```
redirect.url.ips.allowed=
cdn.host.http=http://abc.com:8080/
cdn.dynamic.resources.enabled=false
```

So, even though I don't fully understand it, I'm going to forward so we can get eyes on this soon and then we can try to make sense of this :D